cranelift/codegen/src/ir/pcc.rs

```diff
@@ -434,41 +471,69 @@ impl fmt::Display for Expr {
     }
 }
 
+struct DisplayExprs<'a>(&'a [Expr]);
+
+impl<'a> fmt::Display for DisplayExprs<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self.0.len() {
+            0 => write!(f, "{{}}"),
+            1 => write!(f, "{}", self.0[0]),
+            _ => {
+                write!(f, "{{")?;
+
+                let mut first = true;
+                for expr in self.0 {
+                    if first {
+                        write!(f, " {expr}")?;
+                        first = false;
+                    } else {
+                        write!(f, ", {expr}")?;
+                    }
+                }
+
+                write!(f, " }}")?;
```

Comment:
Seems like this could be cleaned up with https://doc.rust-lang.org/nightly/std/fmt/struct.Formatter.html#method.debug_set

Reply:
I played with this a bit but (i) I want the single-item case to be bare (no braces) to avoid cluttering up the syntax (otherwise e.g. `range(64, {0}, {255})`) and (ii) `debug_set` takes the `Debug` of elements but I want the `Display`; could adapt for both but then it's arguably more and weirder code than the above (to my eyes at least!).